src/rpc/rawtransaction.cpp

```diff
@@ -652,10 +652,16 @@ static RPCHelpMan combinerawtransaction()
     UniValue txs = request.params[0].get_array();
     std::vector<CMutableTransaction> txVariants(txs.size());
 
+    std::optional<Txid> txid;
     for (unsigned int idx = 0; idx < txs.size(); idx++) {
         if (!DecodeHexTx(txVariants[idx], txs[idx].get_str())) {
             throw JSONRPCError(RPC_DESERIALIZATION_ERROR, strprintf("TX decode failed for tx %d. Make sure the tx has at least one input.", idx));
         }
+        if (txid.has_value() && txid.value() != txVariants[idx].GetHash()) {
```

Comment:
For txs with pre-segwit and nested segwit inputs, txids will not match because the scriptSigs will be different. This check is actually more involved as the transactions need to have scriptSigs stripped.